test_pull_requests

```diff
@@ -551,9 +662,65 @@ module Hub
       return pull_request, bot_comment, comments
     end
 
+    # Tries to extract the commit ID for the base repo from the bot
+    # comments by matching the build URL from the jenkins environment
+    # to the build URL in the PR bot comments
+    def base_commit_id_for_merge(pull_request, settings=nil)
+      repo = pull_request['base']['repo']['name']
+      base_commit = nil
+      num_tries=5
+      sleep_time=SLEEP_TIME
+      $stderr.puts "Checking if current base repo commit ID matches what we expect"
+      if ENV['BUILD_URL'].to_s.empty?
+        base_commit = ''
+      else
+        (1..num_tries).each do |i|
+          comment_pattern = /#{CONTENT_RUNNING} +\(#{ENV['BUILD_URL']}\)/
+          bot_comment = get_comment_matching_regex(pull_request['number'], repo, comment_pattern)
```

Comment:
Maybe not necessary in this PR, but since we have the hash and your methods it would furthermore be nice to have shims for stuff like this so we can compartmentalize the regex into one place and never see regex anywhere else in the code